source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCarouselView.mm

```diff
@@ -0,0 +1,253 @@
+@property ACRPageControl *pageControl;
+@property NSMutableArray<UIView *> *carouselPageViewList;
+@property std::shared_ptr<Carousel> carousel;
+
+@end
+
+@implementation ACRCarouselView
+
+-(instancetype) initWithViewGroup:(UIView<ACRIContentHoldingView> *)viewGroup
+                      rootView:(ACRView *)rootView
+                        inputs:(NSMutableArray *)inputs
+               baseCardElement:(ACOBaseCardElement *)acoElem
+                    hostConfig:(ACOHostConfig *)acoConfig
+{
+    self = [super initWithFrame:CGRectZero];
+    self.translatesAutoresizingMaskIntoConstraints = NO;
+    std::shared_ptr<BaseCardElement> elem = [acoElem element];
+    std::shared_ptr<Carousel> carousel = std::dynamic_pointer_cast<Carousel>(elem);
+    std::shared_ptr<HostConfig> config = [acoConfig getHostConfig];
+    self.carouselPageViewList = [[NSMutableArray alloc] init];
+    
+    self.carousel = carousel;
+    
+    
+    ACRColumnView *carouselViewContainer = [[ACRColumnView alloc] initWithStyle:(ACRContainerStyle)carousel->GetStyle()
+                                                        parentStyle:[viewGroup style]
+                                                         hostConfig:acoConfig
+                                                          superview:viewGroup];
+    
+    for(auto carouselPage: carousel->GetItems()) {
+        
+        ACOBaseCardElement *acoElement = [[ACOBaseCardElement alloc] initWithBaseCardElement:carouselPage];
+        
+        UIView * carouselPageView = [[ACRCarouselPageView alloc] render:carouselViewContainer
+                                                               rootView:rootView
+                                                                 inputs:inputs
+                                                        baseCardElement:acoElement
+                                                             hostConfig:acoConfig];
+        carouselPageView.hidden = YES;
+        
+        [self.carouselPageViewList addObject:carouselPageView];
+    }
+    
+    if (self.carouselPageViewList.count == 0) {
+        return self;
+    }
+    
+    UIView * carouselPagesContainerView = [[ACRCarouselPageContainerView alloc] initWithCarouselPageViewList:_carouselPageViewList];
+    
+    self.carouselPageViewList[0].hidden = NO;
+    self.carouselPageViewIndex = 0;
+    
+    std::string selctedPageControlTintColor = config->GetPageControlConfig().selectedTintColor;
+    std::string unselctedPageControlTintColor = config->GetPageControlConfig().unselectedTintColor;
+    
+    ACRPageControlConfig *pageControlConfig = [[ACRPageControlConfig alloc] initWithNumberOfPages:self.carouselPageViewList.count
+                                                                                     displayPages:@7
+                                                                                 selctedTintColor:[ACOHostConfig convertHexColorCodeToUIColor:selctedPageControlTintColor]
```

Comment:
nit: indentation looks off

Reply:
It should be correct all ':' are in same line vertically